src/main/scala/beam/sim/LoggingParallelEventsManager.scala

```diff
@@ -56,4 +72,53 @@ class LoggingParallelEventsManager @Inject()(config: Config) extends EventsManag
         logger.error(s"Method '$what' failed with: ${t.getMessage}. Stacktrace: $st", t)
     }
   }
+
+  private def createDedicatedHandler: Future[Unit] = {
+    logger.info("Running dedicated blocking handler...")
+    Future {
+      // Let's go for now with blocking approach because it's safe CPU and pretty good for
+      // the scenario when it's many writers but only one reader
+      handleBlocking()
+    }(ec)
+  }
+
+  private def handleNonBlocking(): Unit = {
+    // This is going to waste some CPU
+    while (!isFinished.get()) {
```

Comment:
Consider extract the block of code to a method called `wasteCpu` in order to make the comments unecessary